spec/presenters/person_presenter_spec.rb

```diff
@@ -5,27 +5,38 @@
   let(:person) { profile_user.person }
 
   describe "#as_json" do
-    context "with no current_user" do
-      it "returns the user's public information if a user is not logged in" do
+    context "with no current_user (a user is not logged in)" do
+      it "returns the user's basic profile" do
         expect(PersonPresenter.new(person, nil).as_json).to include(person.as_api_response(:backbone).except(:avatar))
       end
+
+      it "returns the user's additional profile if the user has set additional profile public" do
+        person.profile[:public_details] = true
+        expect(PersonPresenter.new(person, nil).as_json[:profile]).to include(*%i(location bio gender birthday))
+      end
+
+      it "doesn't return user's additional profile if the user hasn't set additional profile public" do
+        person.profile[:public_details] = false
+        expect(PersonPresenter.new(person, nil).as_json[:profile]).not_to include(*%i(location bio gender birthday))
+      end
     end
 
     context "with a current_user" do
       let(:current_user) { FactoryGirl.create(:user)}
       let(:presenter){ PersonPresenter.new(person, current_user) }
+      # here private information == addtional user profile, because additional profile by default is private
 
       it "doesn't share private information when the users aren't connected" do
-        expect(presenter.full_hash_with_profile[:profile]).not_to have_key(:location)
+        expect(presenter.as_json[:profile]).not_to have_key(:location)
       end
```

Comment:
This case is no longer complete, it needs to differentiate whether public_details is true, like above.